### ISSUE TITLE
Allow `GitHubActionRoleProps` to specify custom claims

### ISSUE THREAD

**Issue by hertzsprung** [cdklabs/cdk-pipelines-github#739]
Currently, [repos are suffixed with `:*`](https://github.com/cdklabs/cdk-pipelines-github/blob/ce5119fcc6a843a2280abe0e6feb6dcb0be651b5/src/oidc-provider.ts#L163).  This prevents securing hardening filters on GitHub custom claims such as `repo:octo-org/octo-repo:ref:refs/heads/demo-branch` or `repo:octo-org/octo-repo:environment:Production`.

See https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token.

**Comment by kaizencc** [comment]
sure that makes sense. Are you willing to contribute to get your functionality in?

**Comment by hertzsprung** [comment]
Yup, I have this working locally.  Briefly, I've made `repos` optional, and added `subjectClaims?: string[]` to allow arbitrary claims.  I've retained the [`StringLike` condition](https://github.com/cdklabs/cdk-pipelines-github/blob/main/src/oidc-provider.ts#L112) to allow wildcarding for `repos` and `subjectClaims`.

If this sounds sensible I'll go ahead and prepare a PR ready in the next few days.
